src/main/java/com/imsweb/mph/MphUtils.java

```diff
@@ -315,6 +321,21 @@ public Map<String, MphGroup> getAllGroups() {
         return Collections.unmodifiableMap(_groups);
     }
 
+    public Date getDataLastUpdated() {
```

Comment:
Adding @ctmay4 

Technically, the data last updated is an attribute of the provider (it's the date updated for the data of that provider).  You moved the entire logic here, but that means this will return the date for the internal data, no matter what provider is used.   This feels weird. I know Chuck is probably the only one to use different providers, and he probably doesn't care too much about that date, but I think logically, this change is not correct.

You should leave the method in the interface, and the default implementation, and this new method in MphUtils needs to be a call through to the provider. 

That is more logical that way (and it was also a smaller change).

Reply:
That makes sense